builder/main.py

```
@@ -414,6 +415,63 @@ def __fetch_fs_size(target, source, env):
+                    # Ensure parent directories exist
+                    if rel_path.parent != Path("."):
+                        fs.makedirs(rel_path.parent.as_posix(), exist_ok=True)
+                    # Copy file
+                    with fs.open(rel_path.as_posix(), "wb") as dest:
+                        dest.write(item.read_bytes())
+
+        # Write filesystem image
+        with open(target_file, "wb") as f:
+            f.write(fs.context.buffer)
+
+        return 0
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)